packages/legacy-scripting-runner/middleware-factory.js

```diff
@@ -20,35 +20,51 @@ const renderTemplate = (templateString, context) => {
   return _.template(templateString, options)(finalContext);
 };
 
+const getLowerHeaders = (headers) =>
+  Object.entries(headers).reduce((result, [k, v]) => {
+    result[k.toLowerCase()] = v;
+    return result;
+  }, {});
+
+const applyAuthMappingInHeaders = (authMapping, req, authData) => {
+  const lowerHeaders = getLowerHeaders(req.headers);
+  Object.entries(authMapping).forEach(([k, v]) => {
+    const lowerKey = k.toLowerCase();
+    if (!lowerHeaders[lowerKey]) {
+      req.headers[k] = renderTemplate(v, authData);
+    }
+  });
+};
+
+const applyAuthMappingInQuerystring = (authMapping, req, authData) => {
+  Object.entries(authMapping).forEach(([k, v]) => {
+    req.params[k] = req.params[k] || renderTemplate(v, authData);
```

Comment:
Should the two items on the right-hand be switched, so that a user's authData value replaces anything already present in the request?  

`req.params[k] = renderTemplate(v, authData) || req.params[k];`

But maybe that breaks something else, since this is the current/existing behavior

Reply:
> But maybe that breaks something else, since this is the current/existing behavior

Yep, let's keep the current behavior unless we found WB has a different behavior.